x/recovery/keeper/ibc_callbacks.go

```diff
@@ -185,6 +187,18 @@ func (k Keeper) OnRecvPacket(
 		"dest-channel", packet.DestinationChannel,
 	)
 
+	defer func() {
```

Comment:
Cognitive complexity is 22 for this one, your addition only adds one. But the limit remains 7.
This will need a refactor in the future.
Possibly we could move all the refactor to the rest of the sub issues under the linear issue ENG-284